Framework/Algorithms/test/SparseWorkspaceTest.h

```diff
@@ -298,6 +303,150 @@ class SparseWorkspaceTest : public CxxTest::TestSuite {
     }
   }
 
+  // test the interpolation and the error propagation
+  void test_bilinearInterpolateFromDetectorGrid() {
+    using namespace WorkspaceCreationHelper;
+    auto ws = create2DWorkspaceWithRectangularInstrument(1, 2, 7);
+    const size_t sparseRows = 3;
+    const size_t sparseCols = 6;
+    const size_t wavelengths = 3;
+    auto sparseWS = std::make_unique<SparseWorkspaceWrapper>(
+        *ws, wavelengths, sparseRows, sparseCols);
+    for (size_t i = 0; i < sparseWS->getNumberHistograms(); ++i) {
+      auto &ys = sparseWS->mutableY(i);
+      auto &es = sparseWS->mutableE(i);
+      for (size_t j = 0; j < ys.size(); ++j) {
+        ys[j] = static_cast<double>(i);
+        es[j] = std::sqrt(ys[j]);
+      }
+    }
+    auto &grid = sparseWS->grid();
+    double lat = grid.latitudeAt(0);
+    double lon = grid.longitudeAt(0);
+    auto h = sparseWS->bilinearInterpolateFromDetectorGrid(lat, lon);
+    TS_ASSERT_EQUALS(h.size(), wavelengths)
+    for (size_t i = 0; i < h.size(); ++i) {
+      TS_ASSERT_EQUALS(h.y()[i], 0.0)
+      TS_ASSERT_EQUALS(h.e()[i], 0.0)
+    }
+    lat = (grid.latitudeAt(2) + grid.latitudeAt(1)) / 2.0;
```

Comment:
Could this be split into two tests?

Reply:
Yes will do. I had based this rather long test on an existing test called test_interpolateFromDetectorGrid which used the original interpolation method. I will split up both and pull out the workspace creation into a separate function